packages/medusa/src/services/payment-collection.ts

```diff
@@ -75,9 +80,11 @@ export default class PaymentCollectionService extends TransactionBaseService {
       this.paymentCollectionRepository_
     )
 
-    const query = buildQuery({ id: paymentCollectionId }, config)
-
-    const paymentCollection = await paymentCollectionRepository.find(query)
+    let paymentCollection: PaymentCollection[] = []
+    if (paymentCollectionId) {
+      const query = buildQuery({ id: paymentCollectionId }, config)
+      paymentCollection = await paymentCollectionRepository.find(query)
+    }
```

Comment:
findOne executes 2 queries, first one to count the number of rows and then it query again to return the data. This is done internally by TypeORM in case there are relations returning multiple rows and it has to be aggregated to the main object.
I wanted to avoid this